Client/Frontend/Settings/Rewards Internals/RewardsInternalsSharable.swift

```diff
@@ -0,0 +1,120 @@
+    /// Ask the sharable to copy, move, or save the files that will be shared to the file path passed in
+    ///
+    /// The file path will be a temporary directory and will be removed after the user dismisses
+    /// the share sheet.
+    var generator: RewardsInternalsFileGenerator
+    
+    static func == (lhs: RewardsInternalsSharable, rhs: RewardsInternalsSharable) -> Bool {
+        return lhs.id == rhs.id
+    }
+    
+    static let basic = RewardsInternalsSharable(
+        id: "basic",
+        title: Strings.RewardsInternals.sharableBasicTitle,
+        description: Strings.RewardsInternals.sharableBasicDescription,
+        generator: RewardsInternalsBasicInfoGenerator()
+    )
+    
+    static let logs = RewardsInternalsSharable(
+        id: "logs",
+        title: Strings.RewardsInternals.logsTitle,
+        description: Strings.RewardsInternals.sharableLogsDescription,
+        generator: RewardsInternalsLogsGenerator()
+    )
+    
+    static let promotions = RewardsInternalsSharable(
+        id: "promotions",
+        title: Strings.RewardsInternals.promotionsTitle,
+        description: Strings.RewardsInternals.sharablePromotionsDescription,
+        generator: RewardsInternalsPromotionsGenerator()
+    )
+    
+    static let contributions = RewardsInternalsSharable(
+        id: "contributions",
+        title: Strings.RewardsInternals.contributionsTitle,
+        description: Strings.RewardsInternals.sharableContributionsDescription,
+        generator: RewardsInternalsContributionsGenerator()
+    )
+    
+    static let database = RewardsInternalsSharable(
+        id: "database",
+        title: Strings.RewardsInternals.sharableDatabaseTitle,
+        description: Strings.RewardsInternals.sharableDatabaseDescription,
+        generator: RewardsInternalsDatabaseGenerator()
+    )
+    
+    /// The default set of sharables when sharing from the main Rewards Internals screen
+    static let `default`: [RewardsInternalsSharable] = [
+        .basic,
+        .promotions,
+        .contributions
+    ]
+    
+    /// A set of all the available sharables
+    static let all: [RewardsInternalsSharable] = [
+        .basic,
+        .promotions,
+        .contributions,
+    ]
+}
+
+enum RewardsInternalsSharableError: Error {
+    case rewardsInternalsUnavailable
+}
+
+/// A file generator that copies the Rewards ledger database into the sharable path
+private struct RewardsInternalsDatabaseGenerator: RewardsInternalsFileGenerator {
+    func generateFiles(at path: String, using builder: RewardsInternalsSharableBuilder, completion: @escaping (Error?) -> Void) {
+        // Move Rewards database to path
+        do {
+            let dbPath = URL(fileURLWithPath: builder.rewards.ledger.rewardsDatabasePath)
+            let newPath = URL(fileURLWithPath: path).appendingPathComponent(dbPath.lastPathComponent)
+            try FileManager.default.copyItem(atPath: dbPath.path, toPath: newPath.path)
```

Comment:
We don't need to move the db around right?

Reply:
This was used to copy it into zip directory, no longer executed